### PR TITLE
(0.93.0) Use `with_halos=true` default for `JLD2OutputWriter`

### PR DESCRIPTION
**Pull request by glwagner**
I keep getting bit by this, so I think its time to make the change. We'll see if there are any tests that assume the default.

### CODE REVIEW

**Review by navidcy — APPROVED** [review]
lgtm

let's link that issue/PR which we were discussing this?

**Comment by navidcy** [comment]
The Langmuir example was failing because it was loading the arrays with the halos to plot. Switching to using the MakieExt does the job better and cleaner.

File: src/OutputWriters/jld2_output_writer.jl [hunk 1]
```diff
@@ -71,9 +71,7 @@ Keyword arguments
              will save xy-slices of the bottom-most index.
 
 - `with_halos` (Bool): Whether or not to slice halo regions from fields before writing output.
-                       Note, that to postprocess saved output (e.g., compute derivatives, etc)
-                       information about the boundary conditions is often crucial. In that case
-                       you might need to set `with_halos = true`.
+                       Preserving halo region data can be useful for postprocessing.
```
Comment (navidcy) [inline src/OutputWriters/jld2_output_writer.jl:74]:
```suggestion
                       Preserving halo region data can be useful for postprocessing. Default: false.
```

**Comment by glwagner** [comment]
File size in the doctest coming back to slow down development again...

**Comment by navidcy** [comment]
Let’s remove those filesize related doctests!

**Comment by glwagner** [comment]
> Let’s remove those filesize related doctests!

its hard 😭

**Comment by navidcy** [comment]
> > Let’s remove those filesize related doctests!
> 
> its hard 😭

I can do it ;)

**Comment by glwagner** [comment]
Ah but I think this PR was ready

**Comment by glwagner** [comment]
Can we do it in a new PR?

**Comment by glwagner** [comment]
Hmm well I guess its done so lets see if CI passes

**Comment by navidcy** [comment]
oh sure!

**Comment by navidcy** [comment]
> Hmm well I guess its done so lets see if CI passes

damn, I reverted and then I saw this...

**Comment by glwagner** [comment]
feel free to put it back, we have to wait for CI anyways now

**Comment by glwagner** [comment]
Also I think this is a breaking change

**Comment by navidcy** [comment]
> Also I think this is a breaking change

You mean the changing default? Yes, let's bump minor release.

File: docs/src/model_setup/lagrangian_particles.md [hunk 2]
```diff
@@ -108,6 +108,17 @@ Particle properties can be written to disk using JLD2 or NetCDF.
...
 
+```@setup particles
+using Oceananigans
+grid = RectilinearGrid(size=(10, 10, 10), extent=(1, 1, 1));
+n_particles = 10
+x₀ = zeros(n_particles)
+y₀ = rand(n_particles)
+z₀ = -0.5 * ones(n_particles)
+lagrangian_particles = LagrangianParticles(x=x₀, y=y₀, z=z₀)
+model = NonhydrostaticModel(grid=grid, particles=lagrangian_particles)
```
Comment (glwagner) [inline docs/src/model_setup/lagrangian_particles.md:119]:
```suggestion
model = NonhydrostaticModel(; grid, particles=lagrangian_particles)
```

File: docs/src/model_setup/lagrangian_particles.md [hunk 3]
```diff
@@ -108,6 +108,17 @@ Particle properties can be written to disk using JLD2 or NetCDF.
 
 When writing to JLD2 you can pass `model.particles` as part of the named tuple of outputs.
 
+```@setup particles
+using Oceananigans
+grid = RectilinearGrid(size=(10, 10, 10), extent=(1, 1, 1));
+n_particles = 10
+x₀ = zeros(n_particles)
+y₀ = rand(n_particles)
```
Comment (glwagner) [inline docs/src/model_setup/lagrangian_particles.md:116]:
```suggestion
Nparticles = 10
x₀ = zeros(Nparticles)
y₀ = rand(Nparticles)
```

File: docs/src/model_setup/lagrangian_particles.md [hunk 4]
```diff
@@ -108,6 +108,17 @@ Particle properties can be written to disk using JLD2 or NetCDF.
...
+z₀ = -0.5 * ones(n_particles)
```
Comment (glwagner) [inline docs/src/model_setup/lagrangian_particles.md:117]:
```suggestion
z₀ = -0.5 * ones(Nparticles)
```

File: docs/src/model_setup/lagrangian_particles.md [hunk 5]
```diff
@@ -108,6 +108,17 @@ Particle properties can be written to disk using JLD2 or NetCDF.
...
+lagrangian_particles = LagrangianParticles(x=x₀, y=y₀, z=z₀)
+model = NonhydrostaticModel(grid=grid, particles=lagrangian_particles)
+```
+
 ```@example particles
 JLD2OutputWriter(model, (particles=model.particles,), filename="particles", schedule=TimeInterval(15))
```
Comment (glwagner) [inline docs/src/model_setup/lagrangian_particles.md:123]:
```suggestion
JLD2OutputWriter(model, (; particles=model.particles), filename="particles", schedule=TimeInterval(15))
```

**Comment by navidcy** [comment]
We need https://github.com/CliMA/OrthogonalSphericalShellGrids.jl/pull/44 + a new release of OrthogonalSphericalShellGrids.jl otherwise the cyclical dependency blocks Oceananigans from bumping minor version.

**Pull request merged**